hack/build-kubemacpool-podman.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+if [ -z "$ARCH" ] || [ -z "$PLATFORMS" ] || [ -z "$KUBEMACPOOL_IMAGE_TAGGED" ] || [ -z "$KUBEMACPOOL_IMAGE_GIT_TAGGED" ]; then
+    echo "Error: ARCH, PLATFORMS, KUBEMACPOOL_IMAGE_TAGGED, and KUBEMACPOOL_IMAGE_GIT_TAGGED must be set."
+    exit 1
+fi
+
+IFS=',' read -r -a PLATFORM_LIST <<< "$PLATFORMS"
+
+# Remove any existing manifest and image
+podman manifest rm "${KUBEMACPOOL_IMAGE_TAGGED}" 2>/dev/null || echo "Manifest ${KUBEMACPOOL_IMAGE_TAGGED} not found, skipping removal."
+podman manifest rm "${KUBEMACPOOL_IMAGE_GIT_TAGGED}" 2>/dev/null || echo "Manifest ${KUBEMACPOOL_IMAGE_GIT_TAGGED} not found, skipping removal."
+podman rmi "${KUBEMACPOOL_IMAGE_TAGGED}" 2>/dev/null || echo "Image ${KUBEMACPOOL_IMAGE_TAGGED} not found, skipping removal."
+podman rmi "${KUBEMACPOOL_IMAGE_GIT_TAGGED}" 2>/dev/null || echo "Image ${KUBEMACPOOL_IMAGE_GIT_TAGGED} not found, skipping removal."
+
+podman manifest create "${KUBEMACPOOL_IMAGE_TAGGED}"
+
+for platform in "${PLATFORM_LIST[@]}"; do
+    podman build \
+        --no-cache \
```

Comment:
why no cache?

Reply:
It seems --no-cache is recommend approach. Without --no-cache, the build is faster because it reuses previous layers. But this can sometimes cause outdated or incorrect dependencies to be used. Using --no-cache forces a fresh build, making sure all steps in the Dockerfile run again without using old layers. Since we already remove the old manifest before creating a new one, I think --no-cache is the best choice. However, I'm open to other suggestions.

Reply:
imho we can omit it please and use caching, docker caching seems too fundamental to have bugs beside corner cases
we use caching so far and it is fine iiuc
speed is important (of course unless we find in the future it is buggy and then can report / workaround that)

Reply:
Okay sure.